tests/std/tests/P0009R18_mdspan_layout_left_death/test.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <array>
+#include <cstddef>
+#include <mdspan>
+#include <span>
+
+#include <test_death.hpp>
+
+using namespace std;
+
+// TRANSITION, Test Construction From extents_type
+
+void test_construction_from_other_left_mapping() {
+    layout_left::mapping<dextents<int, 1>> mapping1{dextents<int, 1>{256}};
+    // Value of other.required_span_size() must be representable as a value of type index_type
+    layout_left::mapping<dextents<unsigned char, 1>> mapping2{mapping1};
+}
+
+void test_construction_from_other_right_mapping() {
+    layout_right::mapping<dextents<int, 1>> mapping1{dextents<int, 1>{256}};
+    // Value of other.required_span_size() must be representable as a value of type index_type
+    layout_left::mapping<dextents<unsigned char, 1>> mapping2{mapping1};
+}
+
+#pragma warning(push) // TRANSITION, "/analyze:only" BUG?
+#pragma warning(disable : 28020) // The expression '0<=_Param_(1)&&_Param_(1)<=1-1' is not true at this call
+void test_construction_from_other_stride_mapping_1() {
+    using Ext = extents<int, 2, 4>;
+    layout_stride::mapping<Ext> mapping1{Ext{}, array{1, 1}};
+    // For all r in the range [0, extents_type::rank()), other.stride(r) must be equal to
+    // extents().fwd-prod-of-extents(r)
+    layout_left::mapping<Ext> mapping2{mapping1};
+}
+
+void test_construction_from_other_stride_mapping_2() {
+    layout_stride::mapping<dextents<int, 1>> mapping1{dextents<int, 1>{256}, array{1}};
+    // Value of other.required_span_size() must be representable as a value of type index_type
+    layout_left::mapping<dextents<unsigned char, 1>> mapping2{mapping1};
+}
+#pragma warning(pop) // TRANSITION, "/analyze:only" BUG?
+
+void test_stride_function() {
+    layout_left::mapping<extents<int, 3>> mapping;
+    // Value of i must be less than extents_type::rank()
+    (void) mapping.stride(1);

```

Comment:
Suggestion: Although this isn't shadowing, I find that it's best to name variables differently from any Standard identifiers that are involved. Here, I'd like to be able to glance at `mapping` and know that it's always a Standard nested type, not a test variable.